ocs_ci/ocs/external_ceph.py

```diff
@@ -0,0 +1,924 @@
+
+        Returns:
+            dict: Ceph demon stats
+
+        """
+        ceph_demon_counter = {}
+        for demon in self.get_ceph_demons():
+            if demon.role == 'mgr':
+                continue
+            increment = 1  # len(self.get_osd_devices(demon.node)) if demon.role == 'osd' else 1
+            ceph_demon_counter[demon.role] = ceph_demon_counter[demon.role] + increment if ceph_demon_counter.get(
+                demon.role) else increment
+        return ceph_demon_counter
+
+    def get_metadata_list(self, role, client=None):
+        """
+        Returns metadata for demons of specified role
+
+        Args:
+            role(str): ceph demon role
+            client(CephObject): Client with keyring and ceph-common
+
+        Returns:
+            list: metadata as json object representation
+
+        """
+        if not client:
+            client = self.get_ceph_object('client') if self.get_ceph_object('client') else self.get_ceph_object('mon')
+
+        out, err = client.exec_command('sudo ceph {role} metadata -f json-pretty'.format(role=role))
+
+        return json.loads(out.read().decode())
+
+    def get_osd_metadata(self, osd_id, client=None):
+        """
+        Retruns metadata for osd by given id
+
+        Args:
+            osd_id(int): osd id
+            client(CephObject): Client with keyring and ceph-common
+
+        Returns:
+            dict: osd metadata like::
+
+                 {
+                    "id": 8,
+                    "arch": "x86_64",
+                    "back_addr": "172.16.115.29:6801/1672",
+                    "back_iface": "eth0",
+                    "backend_filestore_dev_node": "vdd",
+                    "backend_filestore_partition_path": "/dev/vdd1",
+                    "ceph_version": "ceph version 12.2.5-42.el7cp (82d52d7efa6edec70f6a0fc306f40b89265535fb) luminous
+                            (stable)",
+                    "cpu": "Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz",
+                    "osd_data": "/var/lib/ceph/osd/ceph-8",
+                    "osd_journal": "/var/lib/ceph/osd/ceph-8/journal",
+                    "osd_objectstore": "filestore",
+                    "rotational": "1"
+                 }
+
+        """
+        metadata_list = self.get_metadata_list('osd', client)
+        for metadata in metadata_list:
+            if metadata.get('id') == osd_id:
+                return metadata
+        return None
+
+    def get_osd_container_name_by_id(self, osd_id, client=None):
+        """
+
+        Args:
+            osd_id (int): osd id
+            client (CephObject): Client with keyring and ceph-common:
+
+        """
+        return self.get_osd_by_id(osd_id, client).container_name
+
+    def get_osd_by_id(self, osd_id, client=None):
+        """
+
+        Args:
+            osd_id (int): osd id
+            client (CephObject): Client with keyring and ceph-common:
+
+
+        Returns:
+            CephDemon: daemon object
```

Comment:
`CephDemon: daemon object or None in case the OSD daemon list is empty`

Reply:
done